### PR TITLE
feat: add tngf

### PR DESCRIPTION
**Pull request by blackcat-118**
## Description 
- **Add TNGF** 
- Update NFs hash for bugs fix
- Test Improvement
- Add missing config in test SMF config
- Udpate N3IWF hash and fix TestNon3GPP

### CODE REVIEW

**Comment by andy89923** [comment]
TNGF submodule?

**Comment by andy89923** [comment]
[force_kill.sh](https://github.com/free5gc/free5gc/blob/main/force_kill.sh) should add TNGF as well.

File: test/non3gpp_test.go [hunk 1]
```diff
@@ -333,10 +399,11 @@ func decryptProcedure(ikeSecurityAssociation *context.IKESecurityAssociation, ik
 
 	ikeMessageData, err := ikeMessage.Encode()
 	if err != nil {
-		return nil, errors.New("Encoding IKE message failed")
+		return nil, errors.New("Encoding IKE message failed:" + err.Error())
 	}
 
-	ok, err := handler.VerifyIKEChecksum(ikeSecurityAssociation.SK_ar, ikeMessageData[:len(ikeMessageData)-checksumLength], checksum, transformIntegrityAlgorithm.TransformID)
+	// [TODO] ike.VerifyIKEChecksum is Deprecate
```
Comment (ianchen0119) [inline test/non3gpp_test.go:405]: should remove the TODO comment?

File: test/non3gpp_test.go [hunk 2]
```diff
@@ -386,15 +453,15 @@ func encryptProcedure(ikeSecurityAssociation *context.IKESecurityAssociation, ik
 	if err != nil {
 		return errors.New("Encoding IKE message error")
 	}
-	checksumOfMessage, err := handler.CalculateChecksum(ikeSecurityAssociation.SK_ai, responseIKEMessageData[:len(responseIKEMessageData)-checksumLength], transformIntegrityAlgorithm.TransformID)
+	// [TODO] security.CalculateChecksum deprecated
```
Comment (ianchen0119) [inline test/non3gpp_test.go:456]: should remove the todo comment?

**Comment by ianchen0119** [comment]
Overall, it looks good to me.

File: test/non3gpp_test.go (hunk 1 above)
Comment (andy89923) [reply test/non3gpp_test.go:405]: yes, this should be removed, thanks

**Pull request merged**